server/game/StatusToken.js

```diff
@@ -1,12 +1,46 @@
-const GameObject = require('./GameObject');
+const EffectSource = require('./EffectSource');
+const AbilityDsl = require('./abilitydsl');
 
-class StatusToken extends GameObject {
+class StatusToken extends EffectSource {
     constructor(game, card, isHonored) {
         super(game, isHonored ? 'Honored Token' : 'Dishonored Token');
         this.honored = !!isHonored;
         this.dishonored = !isHonored;
         this.card = card;
         this.type = 'token';
+        this.persistentEffects = [];
+
+        this.applyHonorEffects();
+    }
+
+    applyHonorEffects() {
+        if(!this.card) {
+            return;
+        }
+        let amount = this.honored ? card => card.getGlory() : card => 0 - card.getGlory();
+        let effect = {
+            match: this.card,
+            effect: AbilityDsl.effects.modifyBothSkills(amount)
+        };
+        this.persistentEffects.push(effect);
+        effect.ref = this.addEffectToEngine(effect);
+        this.game.effectEngine.checkEffects();
```

Comment:
I played with this for a while before working out that the way you've done it is indeed correct.  However, I don't think this `checkEffects()` should be here, otherwise you risk updating state-based effects in the wrong place.  Removing it will make a couple of tests fail, but that's because those tests are lazy (they are just calling 'card.dishonor()` instead of finding a real effect to make that game state change).

Reply:
Yeah that was basically it.  I'll try removing it and see if I can fix the tests.

Reply:
The easy fix for the tests is to just call `checkEffects()` after `card.dishonor()`